### PR TITLE
chore: move packaging & isolate service implementations 

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

**Background**

This is an attempt to move services and wiring logic to their end usage.

For example,  having `runtime` package in the core module did not provide any advantage. This package was only used by `flagd` and contained the logic that binds configurations to service implementations. 

Similarly, `flagd-proxy` exposed `sync` service, and having that inside the core module seemed out of place. 

**What's changed**

I have moved `runtime` package to `flagd` & specific service implementations (flag evaluation and sync service) to their end usage. 

**Why ?**

I considered this to be a pre-requisite for https://github.com/open-feature/flagd/issues/1230. If I attempted to add another `sync` implementation to the current core module, it would have made the whole packaging confusing (ex:- we have two sync proto implementations, which do not serve the same purpose. One for flagd-proxy and the other for flagd).

After this refactoring, I could add the flagd's sync service, where it serve buf connect and I can locate that within flagd. 

Besides, we still have a core module with all the sharables. 

Consider the current packaging in the below diagram,

image

* nore - core sync is for flag data sync listeners such as files, K8s ...

### CODE REVIEW

File: test/integration/go.mod [hunk 1]
```diff
@@ -12,60 +12,104 @@ require (
 )
 
 require (
-	buf.build/gen/go/open-feature/flagd/connectrpc/go v1.12.0-20231031123731-ac2ec0f39838.1 // indirect
-	buf.build/gen/go/open-feature/flagd/grpc/go v1.3.0-20231031123731-ac2ec0f39838.2 // indirect
-	buf.build/gen/go/open-feature/flagd/protocolbuffers/go v1.31.0-20231031123731-ac2ec0f39838.2 // indirect
-	connectrpc.com/connect v1.14.0 // indirect
+	buf.build/gen/go/open-feature/flagd/connectrpc/go v1.15.0-20240215170432-1e611e2999cc.1 // indirect
```
Comment (toddbaert) [inline test/integration/go.mod:15]: optional, but if we're doing big renames/refactors, can we perhaps also change this to `tests.e2e`? These are really e2e tests. You will also need to change the makefile and CI commands: https://github.com/open-feature/flagd/blob/517973fa84f0faaa88c12ce06285b30e655860d9/Makefile#L47-L48

**Review by toddbaert — APPROVED** [review]
Yes I think this makes sense. Left an optional recommendation for you to consider.

File: test/integration/go.mod (hunk 1 above)
Comment (toddbaert) [reply test/integration/go.mod:15]: On second thought, maybe we should do this in a PR, so test changes are isolated from refactors, for increase confidence.

Comment (Kavindu-Dodan) [reply test/integration/go.mod:15]: Yes, i think I will defer this to another PR :) This PR only focuses on moving sources to repackage

**Review by odubajDT — APPROVED** [review]

**Review by thisthat — APPROVED** [review]
Well done 👍

**Comment by Kavindu-Dodan** [comment]
Thank you for the reviews @toddbaert @odubajDT @thisthat , I will merge this now :)

**Pull request merged**